### ISSUE TITLE
Feature request: Support watching non-existent files/directories

### ISSUE THREAD

**Issue by nocnokneo** [samuelcolvin/watchfiles#306]
Instead of making the user jump through all the hoops of monitoring the parent directory of the missing file and filtering change notifications for the one specific missing file, just do this automatically or at least have an option to do this automatically.

**Comment by samuelcolvin** [comment]
I don't think this is possible since the underlying OS watching semantics need a directory to exist.

I guess we could add a layer filtering on top in watchfiles, but I've never wanted this.

Happy to review a PR, but since it's already possible and doesn't seem that common a demand, the code would need to be pretty simple for me to accept it.
